src/SearchIndex/index.ts

```diff
@@ -231,6 +233,39 @@ const deleteStaleProperties = async (
   status.deleted += deleteResult.deletedCount === undefined ? 0 : deleteResult.deletedCount;
 };
 
+/**
+ * Reorders keys in the facets object. This assumes that all keys are strings that
+ * are ordered in insertion order.
```

Comment:
It feels a little weird to rely on object keys insertion order for order instead of arrays, but according to this [SO answer](https://stackoverflow.com/questions/5525795/does-javascript-guarantee-object-property-order/38218582#38218582), it seems reliable. Keeping this an object of keys for facets to avoid changing existing search behavior with the current data structure in the db